pytorch_lightning/core/lightning.py

```diff
@@ -914,10 +914,19 @@ def configure_optimizers(self) -> Union[
 
         If you don't define this method Lightning will automatically use Adam(lr=1e-3)
 
-        Return: any of these 3 options:
-            - Single optimizer
-            - List or Tuple - List of optimizers
-            - Two lists - The first list has multiple optimizers, the second a list of LR schedulers
+        Return: any of these 5 options:
+            - Single optimizer.
+            - List or Tuple - List of optimizers.
+            - Two lists - The first list has multiple optimizers, the second a list of LR schedulers.
+            - Dictionary, with an `optimizer` key and (optionally) a `lr_scheduler` key.
+            - Tuple of dictionaries as described, with an optional `frequency` key.
+
+        .. note:: The `frequency` value is an int corresponding to the number of sequential batches
```

Comment:
Consider making this a "Note:\n" instead of ".. note::", just for consistency with the (majority) of the docs.